sdk-cpp/src/do_config.cpp

```diff
@@ -20,23 +20,45 @@ const char* const g_doPluginAptBinName = DO_PLUGIN_APT_BIN_NAME;
 
 static int WriteIoTConnectionStringToConfigFile(const char* value) noexcept
 {
-    try
+    int returnValue = 0;
+    // ptree's exceptions do not provide an error code, and SDK has no logging of its own.
+    // Return specific errors as a workaround.
+    boost::filesystem::path filePath{microsoft::deliveryoptimization::details::GetConfigFilePath()};
+    boost::system::error_code ec;
+    if (boost::filesystem::exists(filePath.parent_path(), ec))
     {
-        boost::property_tree::ptree configTree;
-        configTree.put("ADUC_IoTConnectionString", value);
+        try
+        {
+            boost::property_tree::ptree configTree;
+            configTree.put("ADUC_IoTConnectionString", value);
 
-        // No other configs are used at this time so overwrite the whole file here
-        const std::string& filePath = microsoft::deliveryoptimization::details::GetConfigFilePath();
-        boost::property_tree::write_json(filePath, configTree);
+            // No other configs are used at this time so overwrite the whole file here
+            boost::property_tree::write_json(filePath.string(), configTree);
 
-        return 0;
+            return 0;
+        }
+        catch (const boost::property_tree::ptree_bad_data&)
+        {
+            returnValue = -1;
```

Comment:
Would it be useful to have an enum representing these different errors?

Reply:
Maybe in future when we are writing more configs from the sdk.